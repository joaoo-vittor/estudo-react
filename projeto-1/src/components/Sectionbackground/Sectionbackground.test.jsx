import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';
import { Sectionbackground } from '.';

describe('<Sectionbackground />', () => {
  it('should render with backgroud dark', () => {
    const { container } = renderTheme(
      <Sectionbackground background={true}>
        <h1>Children</h1>
      </Sectionbackground>,
    );
    expect(screen.getByRole('heading')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render with backgroud light', () => {
    const { container } = renderTheme(
      <Sectionbackground>
        <h1>Children</h1>
      </Sectionbackground>,
    );
    expect(screen.getByRole('heading')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render with backgroud dark', () => {
    renderTheme(
      <Sectionbackground background={true}>
        <h1>Children</h1>
      </Sectionbackground>,
    );
    expect(screen.getByRole('heading')).toBeInTheDocument();
  });
});
