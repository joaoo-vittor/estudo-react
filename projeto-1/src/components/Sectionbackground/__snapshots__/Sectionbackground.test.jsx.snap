// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sectionbackground /> should render with backgroud dark 1`] = `
.c0 {
  background: #FFFFFF;
  color: #0A1128;
  background: #0A1128;
  color: #FFFFFF;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0;
  padding: 3.2rem;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h1>
        Children
      </h1>
    </div>
  </div>
</div>
`;

exports[`<Sectionbackground /> should render with backgroud light 1`] = `
.c0 {
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0;
  padding: 3.2rem;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h1>
        Children
      </h1>
    </div>
  </div>
</div>
`;
